pkg/volume/flexvolume/plugin-attacher.go

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package flexvolume
+
+import (
+	"time"
+
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/pkg/util/exec"
+	"k8s.io/kubernetes/pkg/util/mount"
+	"k8s.io/kubernetes/pkg/volume"
+)
+
```

Comment:
Looks like you're missing some methods like `Attach.GetDeviceName()`, `AttachableVolumePlugin.NewAttacher()`, etc .

Add compile-time checks to catch incomplete implementations:
```GO
var _ volume.AttachableVolumePlugin = &flexVolumePlugin{}
var _ volume.Attacher = &flexVolumeAttacher{}
...
```